core/src/main/java/io/micronaut/core/util/CopyOnWriteMap.java

```diff
@@ -154,6 +155,10 @@ private synchronized <R> R update(Function<Map<K, V>, R> updater) {
      */
     public static void evict(Map<?, ?> map, int numToEvict) {
```

Comment:
why the whole eviction logic is implemented that way instead of simply removing random element numToEvitc times checking if there is anything left each time?

Reply:
that way it will be safe and there will be no need to catch etc

Reply:
meh, ignore it, any implementation without locks will result in same problem

Reply:
removing a random element from a map is fairly expensive

Reply:
yep, only simpler way would be removing first/last when size of cache map is to big, but that can create hotspots